js/src/views/Connection/connection.js

```diff
@@ -87,38 +88,65 @@ class Connection extends Component {
     if (needsToken && !isConnecting) {
       return (
         <div className={ styles.info }>
-          <div>Unable to make a connection to the Parity Secure API. To update your secure token or to generate a new one, run <span className={ styles.console }>parity signer new-token</span> and supply the token below</div>
+          <div>
+            <FormattedMessage
+              id='connection.noConnection'
+              defaultMessage='Unable to make a connection to the Parity Secure API. To update your secure token or to generate a new one, run {newToken} and supply the token below'
+              values={ {
+                newToken: <span className={ styles.console }>parity signer new-token</span>
+              } } />
+          </div>
           <div className={ styles.form }>
             <Input
-              label='secure token'
-              hint='a generated token from Parity'
               disabled={ loading }
-              error={ validToken || (!token || !token.length) ? null : 'invalid signer token' }
-              value={ token }
+              error={
+                validToken || (!token || !token.length)
+                  ? null
+                  : (
+                    <FormattedMessage
+                      id='connection.invalidToken'
+                      defaultMessage='invalid signer token' />
+                  )
+              }
+              hint={
+                <FormattedMessage
+                  id='connection.token.hint'
+                  defaultMessage='a generated token from Parity' />
+              }
+              label={
+                <FormattedMessage
+                  id='connection.token.label'
+                  defaultMessage='secure token' />
+              }
               onChange={ this.onChangeToken }
-            />
+              value={ token } />
           </div>
         </div>
       );
     }
 
     return (
       <div className={ styles.info }>
-        Connecting to the Parity Secure API.
+        <FormattedMessage
+          id='connection.connectingAPI'
+          defaultMessage='Connecting to the Parity Secure API.' />
       </div>
     );
   }
 
   renderPing () {
     return (
       <div className={ styles.info }>
-        Connecting to the Parity Node. If this informational message persists, please ensure that your Parity node is running and reachable on the network.
+        <FormattedMessage
+          id='connection.connectingNode'
+          defaultMessage='Connecting to the Parity Node. If this informational message persists, please ensure that your Parity node is running and reachable on the network.' />
       </div>
     );
   }
 
   onChangeToken = (event, token) => {
-    const validToken = /[a-zA-Z0-9]{4}-?[a-zA-Z0-9]{4}-?[a-zA-Z0-9]{4}-?[a-zA-Z0-9]{4}/.test(token);
+    const validToken = /[a-zA-Z0-9]{4}(-)?[a-zA-Z0-9]{4}(-)?[a-zA-Z0-9]{4}(-)?[a-zA-Z0-9]{4}/.test(token);
```

Comment:
I can read and understand it as-is so the bug we introduced jumped out immediately. Since I can see no benefit in crunching apart from making me straining my brain & eyes, I'm not going to crush it.